assignments/arrays.js

```diff
@@ -63,34 +63,118 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
-console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*` );
+
+//console.log(inventory[32]);
+//console.log(Object.values(inventory[32]));
+console.log(`Car 33 is a ${inventory[32].car_year} ${inventory[32].car_make} ${inventory[32].car_model}` );
+//console.log(inventory[32].car_year );
+//console.log(inventory[32].car_make );
+//console.log(inventory[32].car_model );
+//console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*` );
 
 
 
 // ==== Challenge 2 ====
 // The dealer needs the information on the last car in their inventory.  What is the make and model of the last car in the inventory?  Log the make and model into the console.
+
+
 let lastCar = 0;
-console.log();
+//let lastCar = inventory[inventory.length-1];
+//console.log(inventory[inventory.length-1].car_make );
+//console.log(inventory[inventory.length-1].car_model );
+let finCar = `Last car is a ${inventory[inventory.length-1].car_make} ${inventory[32].car_model}`
+
+
+console.log(finCar);
 
 // ==== Challenge 3 ====
 // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
-let carModels = [];
-console.log();
+
+
+
+let carModels = inventory.sort(compare);
+console.log(carModels);
+
+
+function compare(a,b) {
+  if (a.car_model < b.car_model)
+    return -1;
+  if (a.car_model > b.car_model)
+    return 1;
+  return 0;
+}
+
+inventory.sort(function(a,b) {
+    var modelA = a.car_model.toUpperCase();
+    var modelB = b.car_model.toUpperCase();
+    if(modelA<modelB) {
+        return -1;
+    }
+    if (modelA>modelB) {
+        return 1;
+    }
+    return 0;
+});
+
+
+
+
+
+
 
 // ==== Challenge 4 ====
 // The accounting team needs all the years from every car on the lot. Create a new array from the dealer data containing only the car years and log the result in the console.
+for (let i = 0; i < inventory.length; i++) {
+  console.log( inventory[i].car_year );
+}
+
+
+
 let carYears = [];
-console.log();
+function yearOnly() {
+    for (let i = 0; i<inventory.length;i++) {
+        carYears.push(inventory[i].car_year);
+    }
+    
+}
+yearOnly(inventory);
+
+
+console.log(carYears);
 
 // ==== Challenge 5 ====
 // The car lot manager needs to find out how many cars are older than the year 2000. Using the carYears array you just created, find out how many cars were made before the year 2000 by populating the array oldCars and logging it's length.
 let oldCars =[];
-console.log(); 
+
+function less2000() {
+    for(let i = 0; i<carYears.length;i++) {
+if(carYears[i]<2000){
+    oldCars.push(carYears[i])
+        }
+    }
+}
+less2000(carYears);
+
+console.log(oldCars.length);
+
+
+
 
 // ==== Challenge 6 ====
```

Comment:
very close on this.

You're returning all the Audi and BMW car make, but what the question is asking for is for you to return the entire object (so the ID, car make, car model, year, etc). As your answer. 

it can be a bit hard to understand it's explicitly asking you for that type of answer, so just make the quick adjustment to your code to return the rest of the object's info and you'll be good to go.